enginetest/queries/foreign_key_queries.go

```diff
@@ -2228,6 +2228,188 @@ var ForeignKeyTests = []ScriptTest{
 			},
 		},
 	},
+	{
```

Comment:
Missing test case: create a foreign key called `child5_ibfk_1` on another table (maybe child1), then do

```sql
mysql> alter table child5 add foreign key (j) references theparent (i);
ERROR 1826 (HY000): Duplicate foreign key constraint name 'child5_ibfk_1'
```

So MySQL enforces global uniqueness for all constraint names in a database, across all tables, but will only automatically renumber constraints when they exist on the same table.